scripts/Merge_Trajectories.py

```diff
@@ -0,0 +1,190 @@
+            return FramefromLine(line)
+
+
+def checkdata(files):
+    """checks if files are matching one Trajectory"""
+    end_frames = []
+    start_frames = []
+    for file in files:
+        if not isComplete(file):
+            return False
+        start_frames.append(startingFrame(file))
+        end_frames.append(endingFrame(file))
+    for start_frame in start_frames:
+        # a starting frame is considered valid if it is the first frame of the given trajectories
+        # a starting frame is considered valid if it comes after an end frame
+        if start_frame != min(start_frames) and start_frame - 1 not in end_frames:
+            raise IncorrectTrajectoryException(f"the starting frame {start_frame} is not matching the other files")
+        # a starting frame is invalid if it appears more than once
+        if start_frames.count(start_frame) > 1:
+            raise IncorrectTrajectoryException(f"multiple files start with the frame {start_frame}")
+    return True
+
+
+def addDatas(trajecs, output, debug):
+    """will merge all data into the output-file
+      "trajecs" is a list of all Trajectory files"""
+    files = trajecs
+
+    # creates a list with all first Frames
+    first_frames = []
+    for file in files:
+        first_frames.append(startingFrame(file))
+
+    debug_counter = 1
+    debug_length = len(first_frames)
+    # until all files are written
+    while first_frames:
+        # determines the lowest starting frame
+        next_frame = min(first_frames)
+        for i, frame in enumerate(first_frames):
+            # adds all data from that file to the output file
+            if frame == next_frame:
+                if debug:
+                    log.debug(f"writing file {debug_counter}/{debug_length}")
+                write(files[i], output)
+                # removes file and the first line from the lists after the data has been written
+                files.pop(i)
+                first_frames.pop(i)
+                debug_counter += 1
+                break
+
+    log.info(f'new File "{output}" was created.')
+
+
+def write(inputfile, output):
+    """writes all data from the inputfile onto the output file"""
+    with open(inputfile, "r") as input:
+        inputdata = input.readlines()
+    with open(output, "a") as output_file:
+        for line in inputdata:
+            if line != "\n" and not line.startswith("#"):
+                output_file.write(line)
+
+
+def addInfos(trajecs, output):
+    with open(trajecs[0], "r") as temp_trajec:
+        lines = temp_trajec.readlines()
+        counter = 0
+        with open(output, "w") as output_file:
+            temp_line = lines[counter]
+            while temp_line.startswith("#") or temp_line == "\n":
+                if temp_line.startswith("#description:"):
+                    output_file.write(temp_line.strip('\n') + " merged with MergeTool\n")
+                else:
+                    output_file.write(temp_line)
```

Comment:
I don't think you should just copy the existing header to the new output, this may contain more tags such as
`# created by tool xyz`
It would be better if it only writes the header entries it needs , e.g. geometry file, fps and provenance (written by MergeTool)

Reply:
done